go/vt/schemadiff/schema_diff_test.go

```diff
@@ -0,0 +1,657 @@
+			fromQueries: []string{
+				"create table t1 (id int primary key, p int, key p_idx (p));",
+				"create table t2 (id int primary key, p int, key p_idx (p), foreign key (p) references t1 (p) on delete no action);",
+			},
+			toQueries: []string{
+				"create table t1 (id int primary key, q int, key q_idx (q));",
+				"create table t2 (id int primary key, q int, key q_idx (q), foreign key (q) references t1 (q) on delete no action);",
+			},
+			expectDiffs:     2,
+			expectDeps:      1,
+			sequential:      true,
+			impossibleOrder: true,
+		},
+	}
+	hints := &DiffHints{RangeRotationStrategy: RangeRotationDistinctStatements}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.fromQueries == nil {
+				tc.fromQueries = createQueries
+			}
+			fromSchema, err := NewSchemaFromQueries(tc.fromQueries)
+			require.NoError(t, err)
+			require.NotNil(t, fromSchema)
+
+			toSchema, err := NewSchemaFromQueries(tc.toQueries)
+			require.NoError(t, err)
+			require.NotNil(t, toSchema)
+
+			schemaDiff, err := fromSchema.SchemaDiff(toSchema, hints)
+			require.NoError(t, err)
+
+			allDiffs := schemaDiff.allDiffs()
+			allDiffsStatements := []string{}
+			for _, diff := range allDiffs {
+				allDiffsStatements = append(allDiffsStatements, diff.CanonicalStatementString())
+			}
+			assert.Equalf(t, tc.expectDiffs, len(allDiffs), "found diffs: %v", allDiffsStatements)
+
+			deps := schemaDiff.AllDeps()
+			depsKeys := []string{}
+			for _, dep := range deps {
+				depsKeys = append(depsKeys, dep.hashKey())
+			}
+			assert.Equalf(t, tc.expectDeps, len(deps), "found deps: %v", depsKeys)
+			assert.Equal(t, tc.sequential, schemaDiff.HasSequentialExecutionDeps())
+
+			orderedDiffs, err := schemaDiff.OrderedDiffs()
+			if tc.impossibleOrder {
+				assert.Error(t, err)
+				assert.Equal(t, ErrImpossibleDiffOrder, err)
+			} else {
+				require.NoError(t, err)
+			}
+			diffStatementStrings := []string{}
+			for _, diff := range orderedDiffs {
+				diffStatementStrings = append(diffStatementStrings, diff.CanonicalStatementString())
+			}
+			if !tc.impossibleOrder {
+				// validate that the order of diffs is as expected (we don't check for the full diff statement,
+				// just for the order of affected tables/views)
+				require.NotNil(t, tc.entityOrder) // making sure we explicitly specified expected order
+				assert.Equalf(t, len(tc.entityOrder), len(orderedDiffs), "expected %d diffs/entities per %v", len(tc.entityOrder), tc.entityOrder)
+				diffEntities := []string{}
+				for _, diff := range orderedDiffs {
+					diffEntities = append(diffEntities, diff.EntityName())
+				}
+				assert.Equalf(t, tc.entityOrder, diffEntities, "diffs: %v", strings.Join(diffStatementStrings, ";\n"))
+			}
+			for _, diff := range orderedDiffs {
+				s := diff.CanonicalStatementString()
+				// Internal sanity, while we're here: see that the equivalence relation has entries for all diffs.
```

Comment:
FWIW, I think as part of the inclusive language effort we're supposed to avoid sanity-check related language.

Reply:
Changed to `integrity`